fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java

```diff
@@ -31,8 +31,12 @@ public static ConnectContext buildConnectContext() {
         // but QeProcessorImpl::reportExecStatus will check query id,
         // So we must disable report query status from BE to FE
         context.getSessionVariable().setReportSuccess(false);
-        // Always use 1 parallel to avoid affect normal query
-        context.getSessionVariable().setParallelExecInstanceNum(1);
+        int parallel = context.getSessionVariable().getStatisticCollectParallelism();
+        if (null != ConnectContext.get()) {
+            parallel = ConnectContext.get().getSessionVariable().getStatisticCollectParallelism();
+        }
+        context.getSessionVariable().setParallelExecInstanceNum(parallel);
+        context.getSessionVariable().setPipelineDop(1);
```

Comment:
Why set to 1?

Reply:
Avoid pipelien auto ajust instanceNum later